test/python/pulse/test_transforms.py

```diff
@@ -401,12 +407,286 @@ def test_multiple_schedules(self):
+        schedule = schedule.insert(17, instructions.Delay(11, d2), mutate=True)
+
+        sched_grouped = pulse.Schedule()
+        sched_grouped += instructions.Delay(5, d1)
+        sched_grouped += instructions.Delay(7, d0)
+        schedule.append(sched_grouped, mutate=True)
+
+        schedule = transforms.align_left(schedule)
+        reference = pulse.Schedule()
+        # d0
+        reference += instructions.Delay(3, d0)
+        reference += instructions.Delay(7, d0)
+        # d1
+        reference = reference.insert(3, instructions.Delay(5, d1))
+        # d2
+        reference += instructions.Delay(11, d2)
+
+        self.assertEqual(schedule, reference)
+
+    def test_align_left_with_barrier(self):
+        """Test left alignment with a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule.append(directives.RelativeBarrier(d0, d1, d2), mutate=True)
+        schedule = schedule.insert(17, instructions.Delay(11, d2), mutate=True)
+
+        sched_grouped = pulse.Schedule()
+        sched_grouped += instructions.Delay(5, d1)
+        sched_grouped += instructions.Delay(7, d0)
+        schedule.append(sched_grouped, mutate=True)
+        schedule = transforms.remove_directives(
+            transforms.align_left(schedule))
+
+        reference = pulse.Schedule()
+        # d0
+        reference += instructions.Delay(3, d0)
+        reference += instructions.Delay(7, d0)
+        # d1
+        reference = reference.insert(3, instructions.Delay(5, d1))
+        # d2
+        reference = reference.insert(3, instructions.Delay(11, d2))
+
+        self.assertEqual(schedule, reference)
+
+
+class TestAlignRight(QiskitTestCase):
+    """Test right alignment transform."""
+
+    def test_align_right(self):
+        """Test right alignment without a barrier."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        schedule = pulse.Schedule()
+        schedule = schedule.insert(1, instructions.Delay(3, d0), mutate=True)
+        schedule = schedule.insert(17, instructions.Delay(11, d2), mutate=True)
+
+        sched_grouped = pulse.Schedule()
+        sched_grouped.insert(2, instructions.Delay(5, d1), mutate=True)
+        sched_grouped += instructions.Delay(7, d0)
+
+        schedule.append(sched_grouped, mutate=True)
+        schedule = transforms.align_right(schedule)
+
+        reference = pulse.Schedule()
+        # d0
+        reference = reference.insert(1, instructions.Delay(3, d0))
+        reference = reference.insert(4, instructions.Delay(7, d0))
+        # d1
+        reference = reference.insert(6, instructions.Delay(5, d1))
+        # d2
+        reference += instructions.Delay(11, d2)
+        self.assertEqual(schedule, reference)
```

Comment:
Update all these too

Reply:
Why aaren't they all +17 because of line 536?

Reply:
> Update all these too

I disagree as per my comment above.

> Why aaren't they all +17 because of line 536?

align_right rearranges the instructions to happen as far to the right as necessary and only that far. In that case line 536 (since moves) ends at time 11. Hence the times of the channels below.